pkg/ccl/backupccl/show_test.go

```diff
@@ -443,6 +447,12 @@ func TestShowBackups(t *testing.T) {
 		sqlDBRestore.QueryStr(t, `SHOW BACKUP $1 IN $2`, rows[2][0], full),
 		sqlDBRestore.QueryStr(t, `SHOW BACKUP LATEST IN $1`, full),
 	)
+
+	// check that full and remote incremental backups appear
+	b3 := sqlDBRestore.QueryStr(t,
+		`SELECT * FROM [SHOW BACKUP LATEST IN $1 WITH incremental_storage= 'nodelocal://0/foo/inc'] WHERE object_type='table'`, full)
```

Comment:
@dt any tips for using `$2` instead of the hardcoded path here? When I use $2, I get the following error ` pq: failed to cast *tree.Placeholder to string`, which bubbles up from the `planner`'s `TypeAsStringOpts` function.

Reply:
The bug occurs for the `SHOW BACKUP LATEST IN $1 WITH incremental_storage=$2` as well

Reply:
This looks like an unrelated bug in backupccl/show.go in that it is calling optFn() too early, outside fn(). It needs to wait to fn() to call it, which also means it can't pick the result type based on it. I'd leave this as-is for now and file a follow-up issue to address it in a standalone change.